modules/mbedtls/configs/config-tls-generic.h

```diff
@@ -477,13 +477,14 @@
 #if defined(CONFIG_MBEDTLS_PSA_CRYPTO_C)
 #define MBEDTLS_PSA_CRYPTO_C
 #define MBEDTLS_PSA_ASSUME_EXCLUSIVE_BUFFERS
+#define MBEDTLS_PSA_STATIC_KEY_SLOTS
```

Comment:
I'm wondering if in some case we might actually want to allow dynamic key slots in the PSA Crypto core as it used to be... e.g. if the heap is already in use in some application. Could that have some advantage?

Reply:
Yeah this is exactly something I would like to discuss with reviewers.

On one hand I don't like to add too many configurations because any end user willing to use Mbed TLS will then need to dig into a lot of implementation details they might be interested in. On the other hand it's also true that `MBEDTLS_PSA_STATIC_KEY_SLOTS` is mostly interesting to applications which are only using PSA Crypto API[^1], but not TLS or X.509 because these heavily rely on heap memory usage. On top of this, as tried to stress in the description, `MBEDTLS_PSA_STATIC_KEY_SLOTS` pre-allocates all the key slots with a size equal to the maximum asymmetric/symmetric key type supported into the build through `PSA_WANT_xxx` symbols. This means that if in your application you use 1 single 4096-bit RSA key and many 256-bit AES keys, then you'll need all the static key slots to support the RSA key size, which is again a waste of RAM.

Since you also asked about this, I'm starting to think that it might be worth adding a `CONFIG_MBEDTLS_PSA_STATIC_KEY_SLOTS`. Perhaps I can set it as `imply`-ed from `CONFIG_MBEDTLS_PSA_CRYPTO_C` so that the user can freely disable it if they know what they are doing.

[^1]: to be more precise also some PSA Crypto API use heap memory.

Reply:
This seems a really nice option! Thanks for the tip